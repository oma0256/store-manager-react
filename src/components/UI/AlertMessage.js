import React from 'react';
import Alert from 'react-bootstrap/Alert';

const AlertMessage = props => {
  const { variant, message, onDismiss } = props;
  return (
    <Alert dismissible variant={variant} onClick={onDismiss}>
      <p>{message}</p>
    </Alert>
  );
};

export default AlertMessage;
