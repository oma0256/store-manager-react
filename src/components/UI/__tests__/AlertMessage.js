import React from 'react';
import { shallow } from 'enzyme';
import AlertMessage from '../AlertMessage';

describe('Alert Message', () => {
  const onDismissMock = jest.fn();
  const alertMessageProps = { onDismiss: onDismissMock };
  const alertMessage = shallow(<AlertMessage {...alertMessageProps} />);

  it('renders correctly', () => {
    expect(alertMessage).toMatchSnapshot();
  });

  it('calls onDismiss function when clicked', () => {
    alertMessage.simulate('click');
    expect(onDismissMock).toHaveBeenCalled();
  });
});
