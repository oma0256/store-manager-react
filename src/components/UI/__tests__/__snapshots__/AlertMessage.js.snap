// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Alert Message renders correctly 1`] = `ShallowWrapper {}`;
